src/pkg/services/m365/api/users.go

```diff
@@ -193,19 +238,196 @@ func (c Users) GetInfo(ctx context.Context, userID string) (*UserInfo, error) {
+	userInfo.HasMailBox = true
+
+	err = c.allowsExchange(ctx, userID, options)
 	if err != nil {
 		if !graph.IsErrExchangeMailFolderNotFound(err) {
+			logger.Ctx(ctx).Errorf("err getting user's mail folder: %s", err)
+
 			return nil, graph.Wrap(ctx, err, "getting user's mail folder")
 		}
 
+		logger.Ctx(ctx).Infof("resource owner does not have a mailbox enabled")
 		delete(userInfo.DiscoveredServices, path.ExchangeService)
+
+		userInfo.HasMailBox = false
+	}
+
+	userInfo.HasOneDrive = true
+
+	err = c.allowsOnedrive(ctx, userID)
+	if err != nil {
+		err = graph.Stack(ctx, err)
+
+		if !clues.HasLabel(err, graph.LabelsMysiteNotFound) {
+			logger.Ctx(ctx).Errorf("err getting user's onedrive's data: %s", err)
+
+			return nil, graph.Wrap(ctx, err, "getting user's onedrive's data")
+		}
+
+		logger.Ctx(ctx).Infof("resource owner does not have a drive")
+
+		delete(userInfo.DiscoveredServices, path.OneDriveService)
+		userInfo.HasOneDrive = false
+	}
+
+	err = c.getAdditionalData(ctx, userID, &userInfo.Mailbox)
+	if err != nil {
+		return nil, err
 	}
 
 	return userInfo, nil
 }
 
+// verify mailbox enabled for user
+func (c Users) allowsExchange(
+	ctx context.Context,
+	userID string,
+	options users.ItemMailFoldersRequestBuilderGetRequestConfiguration,
+) error {
+	_, err := c.stable.Client().UsersById(userID).MailFolders().Get(ctx, &options)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// verify onedrive enabled for user
+func (c Users) allowsOnedrive(ctx context.Context, userID string) error {
+	_, err := c.stable.Client().UsersById(userID).Drives().Get(ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c Users) getAdditionalData(ctx context.Context, userID string, mailbox *MailboxInfo) error {
+	var (
+		rawURL                     = fmt.Sprintf("https://graph.microsoft.com/v1.0/users/%s/mailboxSettings", userID)
+		adapter                    = c.stable.Adapter()
+		builder                    = users.NewUserItemRequestBuilder(rawURL, adapter)
+		ErrMailBoxSettingsNotFound = clues.New("mailbox settings not found")
+		mailBoundErr               clues.Err
+	)
+
+	newItem, err := builder.Get(ctx, nil)
+	if err != nil && !(graph.IsErrAccessDenied(err) || graph.IsErrExchangeMailFolderNotFound(err)) {
+		logger.Ctx(ctx).Errorf("err getting additional data: %s", err)
+
+		return graph.Wrap(ctx, err, "getting additional data")
+	}
+
+	if graph.IsErrAccessDenied(err) {
+		logger.Ctx(ctx).Infof("err getting additional data: access denied")
+
+		mailbox.ErrGetMailBoxSetting = *clues.New("access denied")
+
+		return nil
+	}
+
+	if graph.IsErrExchangeMailFolderNotFound(err) {
+		logger.Ctx(ctx).Infof("err exchange mail folder not found")
+
+		mailbox.ErrGetMailBoxSetting = *ErrMailBoxSettingsNotFound
+
+		return nil
+	}
+
+	additionalData := newItem.GetAdditionalData()
+
+	mailbox.ArchiveFolder = toString(ctx, additionalData["archiveFolder"], &mailBoundErr)
+
```

Comment:
Nit: The empty lines here feel a little arbitrary.  I think you could squish it all together.
```suggestion
```